src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets

```diff
@@ -58,12 +58,12 @@
     <XunitArguments>$(TargetFileName) $(XunitOptions)</XunitArguments>
 
     <TestProgram Condition="'$(TestHost)'!=''">$(TestHost)</TestProgram>
-    <TestArguments Condition="'$(TestHost)'!=''">$(XunitExecutable) $(XunitArguments)</TestArguments>
+    <TestArguments Condition="'$(TestHost)'!=''">$(XunitExecutable) $(XunitArguments) {XunitTraitOptions}</TestArguments>
 
     <TestProgram Condition="'$(TestHost)'==''">$(XunitExecutable)</TestProgram>
-    <TestArguments Condition="'$(TestHost)'==''">$(XunitArguments)</TestArguments>
+    <TestArguments Condition="'$(TestHost)'==''">$(XunitArguments) {XunitTraitOptions}</TestArguments>
```

Comment:
The xunit traits are set on a per-test-project basis in the RunTestsForProject target.  In the current version we just mash them onto the end of the overall command line once we know what we want to run  We set up basic xunit options for all tests above, then the code coverage target comes in and overrides things like the test host and adds new parameters.  Unfortunately, way that the code coverage command line works, we cannot mash the xunit traits onto the end of the command line:

OpenCover.Console.exe <option cover options> -target:<corerun.exe> -targetargs="xunithost xunitoptions"

The key there is the ending ".  We can easily set up using the CodeCoverage.targets overrides everything through xunitoptions, but until we know the xunit traits that are needed per test, we can't add the ending quote.  Simply mashing the xunit traits onto the end doesn't work, since OpenCover will interpret them as arguments to opencover and not to the target process.  Before we basically used {XunitTraitOptions} and then post-replaced it in the target.

The other option is to add some prefix-post fix command line stuff in there, but that seems just as contrived as replacement